src/pretty/list.rs

```diff
@@ -257,9 +289,14 @@ impl<'a> ListStylist<'a> {
                 inner.enclose(open, close)
             }
         };
-        match self.fold_style {
-            FoldStyle::Never => multi,
-            FoldStyle::Fit => multi.clone().flat_alt(flat()).group(),
+        let fold_style = if self.has_line_comment {
+            FoldStyle::Never
+        } else {
+            self.fold_style
+        };
+        match fold_style {
+            FoldStyle::Never => multi(),
+            FoldStyle::Fit => multi().flat_alt(flat()).group(),
```

Comment:
btw, flat_alt is dangerous. It can be O(2^n) if used recursively. Previously i've removed it in most places. see https://enter-tainer.github.io/typstyle/changelog/#v01112---2024-04-09

Reply:
TBH, this function derives from your `pretty_items`, with the general logic kept. I came to realize that `flat_alt` is not necessary on many occasions not long ago.

Reply:
Consider structures that behave differently in the 'break' style, and we can't decide whether to break the line beforehand: with a trailing comma (such as array), optional parens.
Can we still avoid `flat_alt` in these cases?

Reply:
Maybe we can add an `auto` case with flexible `line()`s, and make the existing `flat_alt` cover fewer docs.

Reply:
> this function derives from your pretty_items

Correct. But since v0.11.12 pretty items is not used very frequently. It is mostly replaced by another function with similar capability. I cannot remember its name though but I think you should have seen it before.

> Can we still avoid flat_alt in these cases?

The idea is that we *can* use flat_alt, but only in leaf nodes.